Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -144,39 +144,41 @@ import Foundation
             configPath = clientToken.configURL.absoluteString
         }
 
-        configurationHTTP?.get(configPath, parameters: BTConfigurationRequest(), shouldCache: true) { [weak self] body, response, error in
+        configurationHTTP?.get(configPath, parameters: BTConfigurationRequest(), shouldCache: true) { [weak self] body, error in
             guard let self else {
                 completion(nil, BTAPIClientError.deallocated)
                 return
             }
 
-            if error != nil {
+            if let error {
                 completion(nil, error)
                 return
-            } else if response?.statusCode != 200 {
```

Comment:
Now we can just check that body is not nil, which matches the [Android implementation](https://github.com/braintree/braintree_android/blob/13140a52d349533d1285d7c0a0257852c3f1cd57/BraintreeCore/src/main/java/com/braintreepayments/api/ConfigurationLoader.kt#L35-L48). Seems like we were checking the status as a way to get a read on the body not being nil, instead we can just check the body (which we weren't doing before).

Reply:
Actually it seems like we are doing something slightly more complicated further up the chain so this isn't working the same way for us. Digging into it now.

Reply:
So this is bizarre `fetchPaymentMethodNonces` makes a `get` call, that subsequently calls `fetchOrReturnRemoteConfig` and the `get` called from `fetchPaymentMethodNonces` relies on body and error both being nil, but the response being 200. 

Our tests may be wrong for this but that seems odd since it's a large number of tests. Based on our tests (the ones failing in this PR) it seems like we actually expect to essentially return a `BTConfiguration` with `nil` JSON when making the call to `fetchPaymentMethodNonces` but in all other cases we expect either `body` or `error` to be non-nil.

If someone wants to tackle unwrapping this while I am on PTO feel free, otherwise I'll pick this back up next open dev.